### DISCUSSION TITLE
improve code block in markdown

### DISCUSSION THREAD

**Discussion by liunux4odoo** [zauberzeug/nicegui#1808] [Ideas / Feature Requests]
inline code and multiline codes in markdown are rendered the same as a block.

1. it should render inline code inplace  like github does, and apply different styles to them.
2. it would be better to give a default language parameter to `ui.markdown` to identify code language types.

**Comment by falkoschindler** [comment]
> inline code and multiline codes in markdown are rendered the same as a block.

Sorry, I don't quite understand. As far as I know they render differently:
````py
ui.markdown('''
    This is some `code`!
    
    ```py
    print('Hello, world!')
    ```
''')
````
<img width="210" alt="Screenshot 2023-10-15 at 15 16 45" src="https://github.com/zauberzeug/nicegui/assets/5767091/d0e68064-c911-4a40-a06e-3bc1d0b83986">

Apart from that I find a language parameter to markdown a bit too specific, since code blocks are only one of many Markdown features. And you can easily add the language to fenced code blocks like in the example above.

**Reply by liunux4odoo** [reply]
Thanks for reply.

My fault, I tried to style the code blocks using an incorrect method that simply replace `<code` to `<div classess><code` in `elements.markdown.apply_tailwind`.

Now I know it is simple to just add styles with `ui.add_head_html`.

Thanks again for the excellent work, nicegui is the best ui framework I met in python.
